src/components/in-app-browser.tsx

```diff
@@ -133,7 +137,11 @@ export const InAppBrowser = ({
       }
 
       if (await detectDeeplinkAndNavigate(nextUrl)) {
-        return;
+        return navigator.goBack();
+      }
+
+      if (await detectDynamicLinkAndNavigate(nextUrl)) {
+        return navigator.goBack();
```

Comment:
Return type of this function is `boolean`
```suggestion
        navigator.goBack();
        return false;
```